shell/components/xen/xen-state.js

```diff
@@ -86,7 +90,7 @@ export default Base => class extends Base {
     // TODO(sjmiles): don't need to create these for default _shouldUpdate
     this._lastProps = Object.assign(nob(), this._props);
     this._lastState = Object.assign(nob(), this._state);
-    this._didUpdate(this._props, this._state, this._lastProps, this._lastState);
+    this._didUpdate(...this._propsAndState());
```

Comment:
This is what 'props and state' always makes me think of:

https://en.wikipedia.org/wiki/Church_and_State_(comics)